libraries/rush-lib/src/pluginFramework/PhasedCommandHooks.ts

```diff
@@ -87,6 +88,23 @@ export class PhasedCommandHooks {
   public readonly createOperations: AsyncSeriesWaterfallHook<[Set<Operation>, ICreateOperationsContext]> =
     new AsyncSeriesWaterfallHook(['operations', 'context'], 'createOperations');
 
+  /**
+   * Hook invoked when operation statue changed
+   * Use the context to distinguish between the initial run and phased runs.
+   * Hook is series for stable output.
+   */
+  public readonly beforeExecuteOperations: AsyncSeriesHook<[Map<Operation, OperationExecutionRecord>]> =
```

Comment:
Favor the `IOperationExecutionResult` interface over the internal `OperationExecutionRecord` class; it's meant to hide some internals and avoid having to export quite as many types, as well as making explicit that some types are meant to be readonly.

Reply:
Done~